### PR TITLE
Clean up debug code

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/debug.cpp [hunk 1]
```diff
@@ -58,32 +55,38 @@ void GiveGoldCheat()
...
-		if (myPlayer.InvGrid[i] == 0) {
-			int ni = myPlayer._pNumInv++;
-			SetPlrHandItem(&myPlayer.InvList[ni], IDI_GOLD);
-			GetPlrHandSeed(&myPlayer.InvList[ni]);
-			myPlayer.InvList[ni]._ivalue = GOLD_MAX_LIMIT;
-			myPlayer.InvList[ni]._iCurs = ICURS_GOLD_LARGE;
-			myPlayer._pGold += GOLD_MAX_LIMIT;
-			myPlayer.InvGrid[i] = myPlayer._pNumInv;
-		}
+	for (int8_t &itemId : myPlayer.InvGrid) {
```
Comment (julealgon) [inline Source/debug.cpp:58]: Can we use `auto`?

File: Source/debug.cpp [hunk 2]
```diff
@@ -58,32 +55,38 @@ void GiveGoldCheat()
...
+
+		myPlayer.RemoveInvItem(itemId);
 	}
 
-	for (int i = 0; i < MAXBELTITEMS; i++) {
-		if (myPlayer.SpdList[i]._itype == ITYPE_GOLD)
-			myPlayer.SpdList[i]._itype = ITYPE_NONE;
+	for (auto &item : myPlayer.SpdList) {
+		if (item._itype == ITYPE_GOLD)
+			item._itype = ITYPE_NONE;
```
Comment (StephenCWills) [inline Source/debug.cpp:89]: Why is there all this code that checks for gold in the player's belt? Can we remove this?

File: Source/debug.cpp [hunk 3]
```diff
@@ -146,20 +149,22 @@ void PrintDebugPlayer(bool bNextPlayer)
...
 	if (player.plractive) {
 		sprintf(dstr, "  Plr %i is %s", dbgplr, player._pName);
 		NetSendCmdString(1 << myplr, dstr);
-		sprintf(dstr, "  Lvl = %i : Change = %i", player.plrlevel, player._pLvlChanging);
+		sprintf(dstr, "  Lvl = %i : Change = %i", player.plrlevel, player._pLvlChanging ? 1 : 0);
 		NetSendCmdString(1 << myplr, dstr);
 		const Point target = player.GetTargetPosition();
-		sprintf(dstr, "  x = %i, y = %i : tx = %i, ty = %i", player.position.tile.x, player.position.tile.y, target.x, target.y);
+		sprintf(dstr, "  x = %i, y = %i : tx = %i, ty = %i", player.position.tile.x,
+		    player.position.tile.y, target.x, target.y);
```
Comment (StephenCWills) [inline Source/debug.cpp:162]:
```suggestion
			player.position.tile.y, target.x, target.y);
```

File: Source/debug.cpp [hunk 4]
```diff
@@ -146,20 +149,22 @@ void PrintDebugPlayer(bool bNextPlayer)
...
 		NetSendCmdString(1 << myplr, dstr);
-		sprintf(dstr, "  mode = %i : daction = %i : walk[0] = %i", player._pmode, player.destAction, player.walkpath[0]);
+		sprintf(dstr, "  mode = %i : daction = %i : walk[0] = %i", player._pmode,
+		    player.destAction, player.walkpath[0]);
```
Comment (StephenCWills) [inline Source/debug.cpp:165]:
```suggestion
			player.destAction, player.walkpath[0]);
```

File: Source/debug.cpp [hunk 5]
```diff
@@ -170,7 +175,8 @@ void PrintDebugQuest()
 {
 	char dstr[128];
 
-	sprintf(dstr, "Quest %i :  Active = %i, Var1 = %i", dbgqst, quests[dbgqst]._qactive, quests[dbgqst]._qvar1);
+	sprintf(dstr, "Quest %i :  Active = %i, Var1 = %i", dbgqst, quests[dbgqst]._qactive,
+	    quests[dbgqst]._qvar1);
```
Comment (StephenCWills) [inline Source/debug.cpp:179]:
```suggestion
		quests[dbgqst]._qvar1);
```

**Review by julealgon — APPROVED** [review]

**Review by StephenCWills — CHANGES_REQUESTED** [review]

File: Source/debug.cpp (hunk 2 above)
Comment (julealgon) [reply Source/debug.cpp:89]: Gold was at some point allowed to go in the belt. There are a few places in the codebase that check this. Would suggest leaving to a separate change that focuses on that aspect alone.

File: Source/debug.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/debug.cpp:162]: Alignment is done with spaces

File: Source/debug.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/debug.cpp]: I think most of it has already been deleted else where.

**Review by StephenCWills — APPROVED** [review]

File: Source/debug.cpp (hunk 2 above)
Comment (StephenCWills) [reply Source/debug.cpp]: @julealgon Did they patch it out, or was it changed before the official release?

**Pull request merged**